fix: correct regex match condition in SyntaxValidator (#2384)
zowe/zowe-cli: packages/imperative/CHANGELOG.md

All notable changes to the Imperative package will be documented in this file.

> ## `8.10.2`

Reviewer: Please use "Recent Changes" for the header. When this PR is merged our automation will update it.
```suggestion
## Recent Changes
```

- BugFix: Broken validation of positional arguments. [#2384](https://github.com/zowe/zowe-cli/pull/2384)

> - Enhancement: Added a test to validate it . [#2384](https://github.com/zowe/zowe-cli/pull/2384)

Reviewer 2: ```suggestion
```
I believe we don't need to add an entry for non-user facing items like a new regression test 😋

> - bugfix: fixed a typo in the syntax validation code for positional arguments. the bug caused the validation to never fail. [#2384](https://github.com/zowe/zowe-cli/pull/2384)
> - enhancement: added a regression test to validate the fix. [#2384](https://github.com/zowe/zowe-cli/pull/2384)  
>   (note: non-user-facing items like regression tests are typically not included in the changelog).

Reviewer: Please use capitalization, link to the issue instead of the PR, and remove the enhancement since changes to tests don't need to be listed.

```suggestion
- BugFix: Fixed a typo in the syntax validation code for positional arguments which caused the validation to never fail. [#2375](https://github.com/zowe/zowe-cli/issues/2375)
```
